src/SqlProcessor.php

```diff
@@ -379,12 +379,15 @@ protected function processMultiValues($type, array $value)
 	{
 		if (empty($value)) {
 			throw new InvalidArgumentException('Modifier %values[] must contain at least one array element.');
-		} elseif (empty($value[0])) {
+		}
+
+		$firstSubValue = reset($value);
+		if (empty($firstSubValue)) {
 			return 'VALUES ' . str_repeat('(DEFAULT), ', count($value) - 1) . '(DEFAULT)';
```

Comment:
This seems incomplete even in the original implementation. We should either trigger error if each row does not have same keys, or support that (filling DEFAULT for all keys missing in that row).

```php
%values[], [
  [],
  ['id' => 1337]
]
```

should produce

```sql
(id) VALUES (DEFAULT), (1337)
```